kusto/ingest/internal/queued/queued.go

```diff
@@ -219,7 +219,7 @@ func (i *Ingestion) Blob(ctx context.Context, from string, fileSize int64, props
 
 	props.Ingestion.RetainBlobOnSuccess = !props.Source.DeleteLocalSource
 
-	err = CompleteFormatFromFileName(&props, from)
+	err = CompleteFormatFromFileName(&props, properties.RemoveQueryParamsFromUrl(from))
```

Comment:
Can we do this only where we return an error or an object where the full URI is not needed?

Reply:
This is not the right way to enforce this.
If you want to do something covering all possible paths, create a SecretiveUri like structure and have String() and SecureString() differ on the output.

Reply:
Not 100% sure

I get the idea, and type safety is good, but I wonder if it can even be done correctly without infecting the whole code and having breaking changes

Reply:
As discussed, I'm going to reduce the scope of this to only the explicit prints.

Afterwards, when I have time, I'll pass an object that contains both a secretive and a sanitized URL, and pass the latter to the methods that don't require the secret.